### PR TITLE
Creation of elg can fail. Handle the error

### PR DESCRIPTION
**Pull request by TingDaoK**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by rccarper** [comment]
I think we have an existing bug with the on_error label here.  I don't expect that we can do reliably do a s3_client_release in this case.

**Comment by TingDaoK** [comment]
> I think we have an existing bug with the on_error label here. I don't expect that we can do reliably do a s3_client_release in this case.

What existing bug? We should probably fix it as well

File: source/s3_client.c [hunk 1]
```diff
@@ -393,9 +398,22 @@ struct aws_s3_client *aws_s3_client_new(
...
-    aws_s3_client_release(client);
-
+    aws_event_loop_group_release(client->body_streaming_elg);
+    client->body_streaming_elg = NULL;
+    if (client->tls_connection_options) {
+        aws_tls_connection_options_clean_up(client->tls_connection_options);
+        aws_mem_release(client->allocator, client->tls_connection_options);
+        client->tls_connection_options = NULL;
+    }
+elg_create_fail:
```
Comment (rccarper) [inline source/s3_client.c:408]: I think this should be moved after the event_loop_group_release?

Comment (TingDaoK) [reply source/s3_client.c:408]: Do, you mean after the `aws_event_loop_group_release(client->client_bootstrap->event_loop_group);`? But, this `elg_create_fail` tag will handle the error when body_streaming_elg failed to create, and the `client_config->client_bootstrap->event_loop_group` is acquired before the creation of body_streaming_elg.

**Review by rccarper — APPROVED** [review]

**Pull request merged**